Sync `SVGMatrix` with WebIDL Specification

Ahmad-S792 opened the pull request:
#### 24a05f37194bf8880e4481306ee3b7db29402ea6
<pre>
Sync `SVGMatrix` with WebIDL Specification
<a href="https://bugs.webkit.org/show_bug.cgi?id=300248">https://bugs.webkit.org/show_bug.cgi?id=300248</a>
<a href="https://rdar.apple.com/162050536">rdar://162050536</a>

Reviewed by Anne van Kesteren.

This patch aligns WebKit with Blink / Chromium, Gecko / Firefox and
Web Specification [1]

[1] <a href="http://www.w3.org/TR/SVG11/coords.html#InterfaceSVGMatrix">http://www.w3.org/TR/SVG11/coords.html#InterfaceSVGMatrix</a>

This PR changes &apos;unrestricted float&apos; / &apos;unrestricted double&apos; to
&apos;float&apos; / &apos;double&apos; as per specification.

* Source/WebCore/svg/SVGMatrix.idl:
* LayoutTests/svg/dom/SVGMatrix.html: Synced from Blink / Chromium
* LayoutTests/svg/dom/SVGMatrix-expected.txt:

&gt; Updated usage from SVGMatrix to DOMMatrix:
* LayoutTests/fast/canvas/canvas-path-addPath.html: Synced from Blink / Chromium
* LayoutTests/fast/canvas/canvas-path-addPath-expected.txt:

Canonical link: <a href="https://commits.webkit.org/301116@main">https://commits.webkit.org/301116@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aa014fa8fd1f5b9e6831d3f6f0fd9a7601c856a7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124937 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44606 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35343 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131787 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/76868 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/2d7ae577-b6db-4083-9503-1bb7b21b60ae) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126814 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45299 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53173 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/95086 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63083 "Unable to confirm if test failures are introduced by change, retrying build") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/3e4185c8-f503-4821-8a78-ec2e6f284e00) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127891 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36143 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111731 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75634 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/ed4612b7-f447-4f0a-a81f-f4df598d55da) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/35078 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29884 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75265 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105905 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30115 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134458 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51767 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39566 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103566 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52188 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107944 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103342 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26320 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48679 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26967 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48786 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51646 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57448 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/51028 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54383 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52721 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/svg/SVGMatrix.idl:
```diff
@@ -20,30 +20,31 @@
     Exposed=Window,
     TaggedWrapper,
 ] interface SVGMatrix {
     // FIXME: these attributes should all be floats but since we implement
     // AffineTransform with doubles setting these as doubles makes more sense.
```
Can this be removed as well?

Ahmad-S792 replied:
I think we can remove it once we move our AffineTransform to float and these bits to float. So unfortunately not at the moment. :-(

annevk replied:
Oh the specification has them as float? Interesting. Carry on.

Ahmad-S792 committed:
Sync `SVGMatrix` with WebIDL Specification
https://bugs.webkit.org/show_bug.cgi?id=300248
rdar://162050536

Reviewed by Anne van Kesteren.

This patch aligns WebKit with Blink / Chromium, Gecko / Firefox and
Web Specification [1]

[1] http://www.w3.org/TR/SVG11/coords.html#InterfaceSVGMatrix

This PR changes 'unrestricted float' / 'unrestricted double' to
'float' / 'double' as per specification.

* Source/WebCore/svg/SVGMatrix.idl:
* LayoutTests/svg/dom/SVGMatrix.html: Synced from Blink / Chromium
* LayoutTests/svg/dom/SVGMatrix-expected.txt:

> Updated usage from SVGMatrix to DOMMatrix:
* LayoutTests/fast/canvas/canvas-path-addPath.html: Synced from Blink / Chromium
* LayoutTests/fast/canvas/canvas-path-addPath-expected.txt:

Canonical link: https://commits.webkit.org/301116@main
